src/jit/codegenarmarch.cpp

```diff
@@ -343,7 +343,11 @@ void CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
         case GT_LABEL:
             genPendingCallLabel       = genCreateTempLabel();
             treeNode->gtLabel.gtLabBB = genPendingCallLabel;
+#if defined(_TARGET_ARM)
+            emit->emitIns_J_R(INS_adr, EA_PTRSIZE, genPendingCallLabel, treeNode->gtRegNum);
```

Comment:
@mskvortsov  I don't know deeply. But I just copied that from the original source of codegenarm.cpp. And then, The assertion has resolved. 
It seems to need more investigate deeply like what you said for using emitIns_R_L insted of emitIns_J_R().